src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -724,6 +726,33 @@ rpmostree_compose_builtin_tree (int             argc,
         goto out;
     }
 
+  if (opt_metadata_json)
```

Comment:
Should we put this *before* handling `opt_metadata_strings`, so that the JSON values can be overridden by `--add-metadata-string`?

Actually, this is even trickier because glib will gladly accept and serialize the same key multiple times, right? From a quick test, `g_variant_dict_lookup` functions will use the latter values, but the earlier values are still accessible through e.g. `g_variant_get_child`. Maybe we should build a `GHashTable` of `s --> v` instead and then serialize that at the end? (That would also fix the hack we currently have in `parse_keyvalue_strings` to fetch out the version string.)

I'm OK with doing this at a later time though if you'd like. (Though maybe for now ordering them the other way around to get override behaviour?)